packages/langium/src/utils/cst-util.ts

```diff
@@ -125,6 +125,7 @@ export function findLeafNodeAtOffset(node: CstNode, offset: number): LeafCstNode
         let firstChild = 0;
         let lastChild = node.children.length - 1;
         while (firstChild <= lastChild) {
+            if (firstChild === lastChild) return findLeafNodeAtOffset(node.children[firstChild], offset);
```

Comment:
Suggestion: change the loop condition to `firstChild <= lastChild` and move this case to after the loop. That would separate the cases more clearly.

I'd also suggest to wrap the `if` body in a block with curly braces for the sake of consistency. I know we often used to write with a more compact style with Xtend, but I see consistency as more important than brevity nowadays.

Reply:
Thanks for the suggestion! I have implemented that (I assumend from context you meant "change the loop condition to `firstChild < lastChild`").